src/main/scala/is/hail/methods/KinshipMatrix.scala

```diff
@@ -0,0 +1,37 @@
+package is.hail.methods
+
+import org.apache.spark.mllib.linalg.Vectors
+import org.apache.spark.mllib.linalg.distributed.{IndexedRow, IndexedRowMatrix}
+import scala.collection.Searching._
+
+/**
+  * Represents a KinshipMatrix, which is a matrix where the number at index (i, j) represents how related the samples at index
+  * i and j in the samples array are.
+  */
```

Comment:
Represents a kinship matrix. Entry (i, j) of the matrix encodes the relatedness of the ith and jth samples in sampleIds.